sdk/log/provider.go

```diff
@@ -41,6 +42,12 @@ func newProviderConfig(opts []LoggerProviderOption) providerConfig {
 
 	if c.resource == nil {
 		c.resource = resource.Default()
+	} else {
+		var err error
+		c.resource, err = resource.Merge(resource.Environment(), c.resource)
```

Comment:
Why not do it in `WithResource`, as for metrics and traces?

Reply:
Good call, I'll move it there 👍